flocker/testtools/test/test_base.py

```diff
@@ -0,0 +1,71 @@
+"""
+Tests for flocker base test cases.
+"""
+
+import errno
+import shutil
+
+from hypothesis import given
+from hypothesis.strategies import binary
+from testtools import TestCase
+from testtools.matchers import (
+    DirExists,
+    Not,
+    PathExists,
+)
+from testtools.testresult.doubles import Python27TestResult
+from twisted.python.filepath import FilePath
+from twisted.trial import unittest
+
+from .._base import AsyncTestCase
+
+
+class AsyncTestCaseTests(TestCase):
+
+    @given(binary())
+    def test_trial_skip_exception(self, reason):
+        """
+        If tests raise the ``SkipTest`` exported by Trial, then that's
+        recorded as a skip.
+        """
+
+        class SkippingTest(AsyncTestCase):
+            def test_skip(self):
+                raise unittest.SkipTest(reason)
+
+        test = SkippingTest('test_skip')
+        result = Python27TestResult()
```

Comment:
First I wondered by Python 2.7 specifically.  Then I wondered why a test double at all.  Looks like the answer to the latter is because this adds "what happened" logging behavior that the stdlib test result doesn't have and the answer to the former is that testtools doesn't offer a "current version of Python" test result double.

I guess Python 2.7 is the explictly supported version of Python for Flocker.  And I guess the chance that we'll switch to another version in the foreseeable future is basically nil.  So this is probably a non-problem.